light/provider/mock/mock.go

```diff
@@ -55,10 +56,16 @@ func (p *Mock) String() string {
 	return fmt.Sprintf("Mock{id: %s, headers: %s, vals: %v}", p.id, headers.String(), vals.String())
 }
 
-func (p *Mock) LightBlock(_ context.Context, height int64) (*types.LightBlock, error) {
+func (p *Mock) LightBlock(ctx context.Context, height int64) (*types.LightBlock, error) {
```

Comment:
Is there a reason that we use handwritten mocks instead of codegen'd ones?

Reply:
Honestly for a lot of the light client tests, we could have used something like mockery but for one reason or another these mocks were made instead. At the end of the day I think they're just as easy to test with

Reply:
Sounds good. Was mostly just wondering if there was a preference. 

In general I prefer mockery gen'd mocks. They give you a lot 'for free', like the ability to assert that things were called with what you expect them to be called with etc. They're also very flexible in that they allow you to return basically whatever you want.

Reply:
I do tend to agree with you and would be happy if you wanted to rip up some of the light client tests and replace them with mockery mocks. We also use the mocks here for benchmarking light client verification speed which I don't think you'd be able to replicate with mockery

Reply:
I might take you up on that.

Reply:
We have both @williambanfield. I don't know why we have hand-written ones. It's probably for testing purposes.